genie-agent/src/main/java/com/netflix/genie/agent/cli/CliAutoConfiguration.java

```diff
@@ -0,0 +1,358 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.agent.cli;
+
+import com.netflix.genie.agent.AgentMetadata;
+import com.netflix.genie.agent.execution.ExecutionContext;
+import com.netflix.genie.agent.execution.services.AgentHeartBeatService;
+import com.netflix.genie.agent.execution.services.AgentJobService;
+import com.netflix.genie.agent.execution.services.DownloadService;
+import com.netflix.genie.agent.execution.services.KillService;
+import com.netflix.genie.agent.execution.statemachine.JobExecutionStateMachine;
+import com.netflix.genie.proto.PingServiceGrpc;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Lazy;
+
+import javax.validation.Validator;
+
+/**
+ * Spring auto configuration class to contain all beans involved in the CLI for the Agent.
+ *
+ * @author tgianos
+ * @since 4.0.0
+ */
+@Configuration
+public class CliAutoConfiguration {
+
+//    @Bean
```

Comment:
Should this be removed?

Reply:
It's commented back in / modified in another commit later in this PR